lib/trailblazer/rails/controller.rb

```diff
@@ -10,9 +10,15 @@ def run_v21(operation, *dependencies, **variables)
 
       _assign_trb_ivars(result)
 
-      yield(result) if result.success? && block_given?
-
       @_result = result
+
+      return unless result.success? && block_given?
```

Comment:
I'm not sure this tests what you intend. Maybe this should be `return if result.failure?`, only? We don't care about the block, anyway, in this case.